lib/handler/configurator/proxy.c

```diff
@@ -89,14 +89,13 @@ static SSL_CTX *create_ssl_ctx(void)
     return ctx;
 }
 
-static h2o_cache_t *create_ssl_session_cache(size_t capacity, unsigned lifetime)
+static h2o_cache_t *null_session_cache = (h2o_cache_t *){0}; /* indicate to make session cache empty */
```

Comment:
I am very opposed to this. It is so easy to misuse a special value like `null_session_cache`.

As an alternative approach, how about passing `h2o_cache_t **session_cache` as an argument to `update_ssl_ctx`? If `session_cache` is NULL, then the function inherits the current parameters. Otherwise, session cache of the updated SSL_CTX is set to the value of `*session_cache`.